pkg/config/validation.go

```diff
@@ -0,0 +1,2490 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
```

Comment:
`pkg/config/validation`?

Reply:
@jasonwzm 
Jason is also working on a Cue based model for code-genning validation Go code from Cue input. If the fidelity of this is good enough, we maybe able to code-gen directly into an isolated package. Having an isolated package to host validation related code may help.

Reply:
+1

Reply:
Yes, the validation code here will be replaced by generated code.